src/include/simeng/OS/SyscallHandler.hh

```diff
@@ -33,6 +34,44 @@ static constexpr uint16_t PATH_MAX_LEN = 4096;
 namespace simeng {
 namespace OS {
 
+/** Enum representing the status of a process that has invoked the futex
+ * syscall. */
+enum class FutexStatus : uint8_t { FUTEX_SLEEPING, FUTEX_AWAKE };
+
+/** This struct stores all information required to perform the futex syscall.*/
+struct FutexInfo {
+  /** This is the address in memory where the futex word is stored. */
+  uint32_t faddr = 0;
+
+  /** This is the process which invoked the futex syscall. */
+  std::shared_ptr<Process> process = nullptr;
+
```

Comment:
Given this holds a shared ptr reference to the process, please add functionality to the terminate functions inside SimOS to remove this process reference when it is terminated

Reply:
done